cmd/legacy/main.go

```diff
@@ -127,14 +137,55 @@ func main() {
 		if err != nil {
 			logger.Fatal(err.Error())
 		}
-		for _, date := range dates {
-			if !date.IsDir() {
+
+		// Reverse order (so we convert the latest data first)
+		slices.Reverse(dates)
+
+		// Track longest slice and append to task list / map
+		if len(dates) > maxIfaceEntries {
+			maxIfaceEntries = len(dates)
+		}
+		ifaceTasks[iface.Name()] = dates
+	}
+
+	// Loop over all entries in all directories, one directory / timestamp index at a time
+	for i := 0; i < maxIfaceEntries; i++ {
```

Comment:
This may seem a bit crude, but it ensures that data from all interfaces is nicely interleaved so that data is filled more or less in backwards chronological order. In theory you could start `goProbe v4` as soon as the first iteration of `i` is through (because then the currently active directory is present for _all_ interfaces, allowing `goProbe` to hook / write to that.